### ISSUE TITLE
Possibility to add headers ordering in the render method

### ISSUE THREAD

**Issue by moumoutte** [mjumbewu/django-rest-framework-csv#27]
Actually, it's pretty difficult to maintain a renderer with ordering columns when the columns are generated dynamicaly. It will be nice to have a way to override the headers attributes on the  single instance at the render time.

    def render(self, headers=None, *args, **kwargs):
        if headers:
            self.headers = headers
         # Do the rendering

With this patch , we can have two way to fill up the headers values, it will be very helpful.

If you want I can write a patch

**Comment by debnet** [comment]
Please use OrderedDict for flattened lists and dicts, it keeps the order given by DRF.

An option in order to provide the headers' order will be gratefully accepted too!

**Comment by freyley** [comment]
:+1: 

I also could really use ordering based on the serializer field definition.

**Comment by tpeaton** [comment]
:+1: for this feature.

**Comment by moumoutte** [comment]
I'll try a patch as soon as I have some time. Before the week-end I guess

**Comment by travisbloom** [comment]
:+1: was this ever implemented @moumoutte? would be really useful :)

**Comment by moumoutte** [comment]
Just add the order kwargs in render method, does it looks good to you ?

**Comment by mjumbewu** [comment]
Thanks @mamoutte for the pull request! It looks good, but I'm wondering whether it would be more useful if the `order` argument were an attribute on the class instead. I imagine myself using something like this in the following way:

```python
class MyUserRenderer (CSVRenderer):
    order = ['first', 'last', 'email']

@api_view(['GET'])
@renderer_classes((MyUserRenderer,))
def my_view(request):
    users = User.objects.filter(active=True)
    content = [
        {'first': user.first_name, 'last': user.last_name, 'email': user.email}
        for user in users]
    return Response(content)
```

As a backup, it could be passed in with the `renderer_context`:

```python
class MyView (APIView):
    renderer_classes = [CSVRenderer]

    def get_renderer_context(self):
        context = super().get_renderer_context()
        context['order'] = (
            self.request.GET['fields'].split(',') 
            if 'fields' in self.request.GET else None)
        return context

    ...
```

```bash
curl http://localhost:8000/api/my_view?fields=first,last,email
```

Either way, I feel like I don't call `render` directly from my code, so the keyword argument would be of limited use to me. Am I an outlier in that regard?

**Comment by freyley** [comment]
100% agree with mjumbewu about how I would use this.

**Comment by mjumbewu** [comment]
Actually, somehow I missed this, but it seems that what I described above (ordering with a class attribute) was introduced by @robguttman in pull request #3, commit 93236a0b2 (the attribute is called `headers`).

**Comment by mjumbewu** [comment]
I just released 1.4.0 which allows sorting fields using the `header` attribute on the `CSVRenderer` class, or using the `"header"` key in the `renderer_context`. I'm going to close this issue. Thanks @moumoutte for kicking off the thread!

**Comment by debnet** [comment]
It will also be good to override headers with a dict providing custom titles, is it possible?

**Comment by mjumbewu** [comment]
This is actually possible as of version 1.4.1 (released today! see pull request #39)
